### ISSUE TITLE
std::pair treated as an array instead of key-value in `std::vector<std::pair<>>`

### ISSUE THREAD

**Issue by xsacha** [nlohmann/json#1520]
- What is the issue you have?
I have a variable with std::vector<std::pair<std::string, std::string>> that cannot be deserialised by nlohmann::json.

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

JSON: { "options": [{"video_size": ""}, {"framerate": "0"}, {"vcodec": ""}] }

`std::vector<std::pair<std::string,std::string>> m_Options;`
Non-working Code:
`j["options].get_to(m_Options);`
Working code:
```
for (auto &i : j["options"].get<std::vector<std::map<std::string, std::string>>>())
{
    m_Options.push_back(*i.begin());
}
```

- What is the expected behavior?
Should be able to deserialise to an STL type automatically

- And what is the actual behavior instead?
Failed to deserialise because wrong type

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?
MSVC 2017

- Did you use a released version of the library or the version from the `develop` branch?
V3.2.0 - V3.5.0 exhibit this issue

Notes: When serialising m_Options I get:
JSON: { "options":[["video_size",""],["framerate","0"],["vcodec",""]] }

Basically, it treats a std::pair as an array with two elements, rather than a key-value pair.

**Comment by nlohmann** [comment]
I can confirm your observations. We currently can only convert arrays to pairs:

```cpp
template<typename BasicJsonType, typename A1, typename A2>
void from_json(const BasicJsonType& j, std::pair<A1, A2>& p)
{
    p = {j.at(0).template get<A1>(), j.at(1).template get<A2>()};
}
```

Your case would be possible in principle, but it is a very special usecase to have a list of objects where each object has only one entry.

**Comment by xsacha** [comment]
I know it doesn't make much sense because it should just be a map but I'm trying to support this specific STL type that was previously serialised like this in a different json library.
I think it should support going to/from a pair as a key-value.
